pkg/storage/factory.go

```diff
@@ -401,8 +403,11 @@ func NewIndexClient(periodCfg config.PeriodConfig, tableRange config.TableRange,
 			return nil, err
 		}
 
-		shipper, err := shipper.NewShipper(cfg.BoltDBShipperConfig, objectClient, limits,
-			ownsTenantFn, tableRange, registerer, logger)
+		var filterFn downloads.TenantFilter
```

Comment:
```suggestion
		filterFn := ownsTenantFn
```
(if shardingStrategy is nil you'll be passing a nil function while in the current implementation it would pass `ownsTenantFn`)

Reply:
The current implementation also passes `nil` for the `ownsTenantFn` type